### PR TITLE
Port DAO Examples from foam 1 to foam 2 syntax.

### PR DESCRIPTION
**Pull request by drish**
This PR aims to port almost all of [these](http://foam-framework.github.io/foam/foam/demos/dao.html) examples,  from foam version 1 , to foam 2.

This document should be used as a quick reference on how to manipulate data on f2, how mlang works, and overall DAOs operations.

For in depth documentation and implementation details the actual documentation (which atm is `npm run doc`) should be used.

Examples not included in this PR:
  - Relationships
  - Pie graphs
  - u2 UI changes, imo UI examples should be in a separated examples file I guess ?

going to work on more improvements to this doc on future prs.

### CODE REVIEW

File: doc/guides/DaoExamples.md [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+// give me all gibsons or fenders using IN
+guitarDAO.where(expr.IN(Guitar.BRAND, ['gibson', 'fender']))
+.select()
+.then(function(guitars) {
+  console.log(guitars.a.length);
+});
+```
+
+```js
+// give me all fenders with a price below 2500
```
Comment (kgrgreer) [inline doc/guides/DaoExamples.md:173]: The query doesn't specify fenders

File: doc/guides/DaoExamples.md [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+```js
+// give me all fenders and order by price
+var sink = foam.dao.ArraySink.create();
+
+guitarDAO.where(expr.EQ(Guitar.BRAND, 'fender'))
+.orderBy(expr.DESC(Guitar.PRICE))
+.select(sink)
+.then(function() {
+  var a = sink.a;
+  a.forEach(function(g) { console.log(g.price) }) // 2700, 2400, 2000
```
Comment (kgrgreer) [inline doc/guides/DaoExamples.md]: This isn't natural. Instead, you would just process each row in the select:

select({ put: function(g) { console.log(g.price); }})

File: doc/guides/DaoExamples.md [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+guitarDAO.limit(2)
+.select()
+.then(function(g) {
+  console.log(g.a.length);
+});
+```
+
+```js
+// give me only two guitars
+guitarDAO.limit(2)
```
Comment (kgrgreer) [inline doc/guides/DaoExamples.md:224]: This is a duplicate of the previous test.

File: doc/guides/DaoExamples.md (hunk 2 above)
Comment (drish) [reply doc/guides/DaoExamples.md]: yeah, makes sense. didn't know much about sinks when ported this one. 
rewrote this

**Pull request merged**